### ISSUE TITLE
No audio on videorecord

### ISSUE THREAD

**Issue by lodgerz** [libretro/RetroArch#13580]
## Description
hello, video recordings files have no audio.
i have also tried this custom configuration but still no audio:

vcodec = libx264
acodec = libmp3lame
pix_fmt = yuv420p
scale_factor = 2
threads = 3
video_crf = 25
video_preset = superfast
video_tune = animation
audio_global_quality = 75
sample_rate = 44100
format = flv

### Version/Commit

- RetroArch: retroarch 1.9.14-2

### Environment information

- OS: ArchLinux with pulse

**Comment by hizzlekizzle** [comment]
yeah, this is unfortunately a known issue. FFmpeg deprecated/broke the audio recording API calls we relied on.

Our options would be to vendor an older version that still worked or update our implementation to work with the new calls. I believe we're planning to do the latter, since the new 5.0 release is supposed to be LTS.

**Comment by lodgerz** [comment]
> yeah, this is unfortunately a known issue. FFmpeg deprecated/broke the audio recording API calls we relied on.
> 
> Our options would be to vendor an older version that still worked or update our implementation to work with the new calls. I believe we're planning to do the latter, since the new 5.0 release is supposed to be LTS.

ok thanks for reply. good job!

**Comment by keithbowes** [comment]
AFAICT, the only deprecated things left in (after the PR was merged) are:
1. Things necessary for FFmpeg 3 support (namely, the `av_register_all` and `avformat_network_init` functions, and AVFormatContext.filename), but these are in ifdef blocks, and so aren't compiled in with FFmpeg 4 or 5.
2. The AVDictionary structure.  The entire dict.h header is marked as deprecated in FFmpeg 5 (the recommended replacement being the faster tree API in tree.h), but it's required by `avcodec_open2`, so removing it is a no-go at this time.
